File: src/events/mention.js
const createEmbed = require('../utils/embed');

module.exports = (message) => {
  const embedUrl = `https://discordapp.com/oauth2/authorize?&client_id=${message.client.user.id}&scope=bot&permissions=2112`;

  const embed = createEmbed(message.client, 'Ba<3', message.author)
    .setDescription('Ba Nya Awoo Choo Choo Arf Caw Beary')
    .addField('Github Link (PRs encouraged)', 'https://github.com/flutterflies/ba')
    .addField('Invite Link', embedUrl);

  message.channel.send(embed);
};

File: bot/lib/ba/events/mention.rb
# frozen_string_literal: true

require 'ba/utils/embeds'

module Ba
  module Events
    module Mention
      def self.on_mention(event)
        title = 'BAAAAAAAAAAA!!!'
        description = 'Ba Nya Awoo Choo Choo Arf Caw Beary'
        fields = [
          { name: 'Github (PRs encouraged)', value: 'https://github.com/flutterflies/ba' },
          { name: 'Invite Link', value: INVITE_URL }
        ]

        Ba::Utils::Embeds.send_embed(event: event, title: title, description: description, fields: fields)
      end
    end
  end
end
